Fix Local pickup e2e tests

tarunvijwani opened the pull request:
This PR fixes the following failed e2e tests:
- User does not see shipping rates until full address is entered
- Local Pickup Settings › Location Settings › can add a new location
- Local Pickup Settings › Location Settings › can delete a location

We removed the `location_state` select input and changed `location_country` to `location_country_state` select input in #8408 that was causing the e2e to fail. 

## Changes in the PR
- Change `location_country` select input to `location_country_state` select input in e2e tests.
- Remove `location_state` select input in e2e tests.

<!-- Don't forget to update the title with something descriptive. -->
<!-- If you can, add the appropriate labels -->

#### Accessibility

<!-- If you've changed or added any interactions, check off the appropriate items below. You can delete any that don't apply. Use this space to elaborate on anything if needed. -->

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

#### Other Checks

- [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.



### Testing

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

#### Testing

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->


1. Run `npm run test:e2e -- tests/e2e/specs/merchant/local-pickup.test.ts`
2. Confirm all the tests are getting passed. 
3. Run `npm run test:e2e -- tests/e2e/specs/shopper/cart-checkout/checkout.test.js`
2. Confirm all the tests are getting passed. 

* [x] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

Commit:
Fix local pick-up e2e tests

- Change location_country select input to location_country_state select input in e2e tests.
- Remove location_state select input in e2e tests.

Commit:
Merge branch 'trunk' into fix/local-pickup-e2e-tests

Commit:
Fix country/state value

- Change country/state value from 'US' to 'United States (US) — California'

Commit:
Merge remote-tracking branch 'origin/fix/local-pickup-e2e-tests' into fix/local-pickup-e2e-tests

nielslange commented on tests/e2e/specs/merchant/local-pickup.test.ts:
```diff
@@ -254,13 +254,9 @@ describe( `Local Pickup Settings`, () => {
 			);
 			await page.select(
-				'.components-modal__content select[name="location_country"]',
+				'.components-modal__content select[name="location_country_state"]',
```
Nitpick: I'd probably add a note why we're using the selector `location_country_state` with `United Kingdom (UK)`. Before checking #8408, this confused me a bit. After checking the other PR, it was clear to me why we're using this. I believe an inline comment can prevent that others from being confused when reading over this section.

nielslange reviewed (approved):
Thanks for working on this, @tarunvijwani! I'm pre-approving this PR, as the changed E2E tests passed as expected:

<img width="677" alt="Screenshot 2023-03-15 at 11 45 01" src="https://user-images.githubusercontent.com/3323310/225209055-9480f0dd-8992-4ae9-bfed-be19ad398285.png">

<img width="711" alt="Screenshot 2023-03-15 at 11 46 46" src="https://user-images.githubusercontent.com/3323310/225209278-a9ebe2f3-9d18-4e2b-8f6c-91317104aaaf.png">

I just left one minor suggestion regarding leaving a comment to avoid potential confusion regarding `location_country_state` and `United Kingdom (UK)`.

Commit:
Add inline comment about same field for Country and State fields

- We merged the Country and State fields into the same field in the local pickup modal. Add inline comment in e2e test to avoid confusion.
